### DISCUSSION TITLE
I/O Expander ports backward

### DISCUSSION THREAD

**Discussion by MeanGreen420** [theyosh/TerrariumPI#913] [Q&A] [answered]
Hello, Ive been trying to add a PCF8574 8 ch I/O, I finally figured out that the output that responds is opposite the one that was sent the command,  am I doing something wrong?

**Comment by MeanGreen420** [comment]
PCF8574-3,0x20,1 is the address structure im using and in this example io 3 should trigger but instead i get 5, 1 is 7, 8 is 0. If this is how its supposed to be im ok with it.  Should i make a bug report?

**Reply by theyosh** [reply]
Euh, I do not know. I have to look into this. As there is also a different way of using this expander board.

Give me some day to think about this, and how I can test this.

**Reply by theyosh** [reply]
Sorry for the slow response, but I thought I had this hardware at home. But I do not have it. So I will get one tomorrow and do some testing.

**Reply by theyosh** [reply]
The device I bought is maybe not correct. So I ordered a few more from different suppliers. I got strange effects...

**Reply by MeanGreen420** [reply]
I can run more tests on my units if you need. mine works consistent for outputs or inputs,  Thanks

**Comment by theyosh** [comment]
I am so sorry for another bug delay, but I did had time to test some stuff. So I was wondering, how good is your technical skills 

As I am looking at this software package to use now: https://pypi.org/project/i2crelay/

So if you are using a manual installation, and able to switch branches, I can change the code in the `development` branch, and test if the above library works better.

**Comment by theyosh** [comment, marked as answer]
I made a new release, which I changed some libraries and code. For me it worked like expected now.

**Comment by MeanGreen420** [comment]
Ok thanks, i was woundering when i saw the latest update, sorry i did t
reply earlier.  Ill give it a try!

**Reply by theyosh** [reply]
No problem. Just give it a try. I hope it works for you now also

**Reply by MeanGreen420** [reply]
well so far the gpio change affected my system because all the relays were inverse, and i cant get the calibrate option to show up to reverse them. I also lost control of the io exp also, I'm on P1 and have tried 0,1,2,5.  it was working on 7 but doesn't anymore. Ill do some more testing tomorrow.

**Reply by theyosh** [reply]
Make sure you browser cache is cleared. That could help showing the calibration.
